### ISSUE TITLE
delete 'master' branch and just make 'gh-pages' the default branch

### ISSUE THREAD

**Issue by jgravois** [Esri/developer-support#249]
> disclaimer: i just finished cleaning up another merge conflict to get `master` and `gh-pages` and sync.  

my original plan was to follow up and write documentation to explain to people some do's and donts for managing multiple branches and keeping them synchronized, but i think it'd be ~3x easier to just get rid of the `master` branch entirely and make `gh-pages` our default branch.

## what is `gh-pages` anyway?

naming a branch `gh-pages` is just a handy shortcut provided by github to let them know that we'd like them to be start serving the content in the branch as a static webpage.

documentation: [What is Github pages?](https://help.github.com/articles/what-is-github-pages/)

source code:
https://github.com/Esri/developer-support/blob/gh-pages/web-leaflet/browserify/index.html
live website:
http://esri.github.io/developer-support/web-leaflet/browserify/index.html

if a file is not present in the `gh-pages` branch, it is not possible to run it as a live sample.  the fact that we encourage folks to include a [Live Sample](http://esri.github.io/developer-support/web-leaflet/browserify/index.html) link in their README is another handy shortcut, but it doesn't provide any magic.

## but what about samples that have nothing to do with the web?

the good news is, they are perfectly happy to live in a branch named `gh-pages` too!

## what will we have to do differently?

basically, we'll keep doing our own work in unique feature branches (named: `patch` or `my-cool-new-sample`) and start creating pull requests to compare them with `gh-pages` whether they are a website, an ArcObjects code snippet or a Python script.  

thats it.

## why were we ever maintaining two different branches if they are supposed to be identical?

i have no idea.

i've created an 'issue' for this because i am looking for feedback from anyone and everyone here.  my personal opinion is that we are doing more harm than good by asking folks to try and keep two different authoritative branches in sync, but i am *not the boss*.

**Comment by afili** [comment]
@jgravois so are we talking about removing the master completely and replacing with gh-pages instead? So whenever we submit a pull request, we will directly compare with gh-pages, is this correct? If yes, I like this idea. Keeping both was always confusing to me.

**Comment by jgravois** [comment]
> are we talking about removing `master` completely and replacing with `gh-pages` instead?

yes.  precisely.

**Comment by NoashX** [comment]
I'm all in on this idea as well. I've pretty much abstained from touching gh-pages due to confusion on what it is and how it works, so :thumbsup: for me.

**Comment by phpmaps** [comment]
@jgravois @afili @NoashX  - The reasoning that we encourage folks to include a Live Sample is a good rational for me.  To get the live sample link it's always a matter of doing 2 pull requests, which I've not been a fan of.  

I am behind the idea too especially since it wont disrupt non-web based code.

However, does making this change have downstream effects on finding code?  I think every time I browse GitHub projects the default is "Master".  What's to become of "Master" and how do we make people browsing code aware that they need to switch to "gh-pages" within developer-support?  

Thoughts?

**Comment by benstoltz** [comment]
@phpmaps The default branch can be set to gh-pages.

**Comment by jgravois** [comment]
@benstoltz is right.  the repo below is a good example because its already set up in this way

https://github.com/Esri/geodev-hackerlabs

**Comment by phpmaps** [comment]
@benstoltz @jgravois @afili @NoashX  - sweet lets do it.  Great suggestion..

**Comment by jgravois** [comment]
done and done.

fair warning, this will not automatically remove the `master` branch from existing forks or set `gh-pages` as the default branch for forks.  

there is no harm in this, but for the folks out there as OCD as me.  you can remove your own `master` branch manually using github's web tools or by executing the command below (**after** you've set `gh-pages` as the new default branch of your fork).

```bash
$ git push origin :master
```

http://matthew-brett.github.io/pydagogue/gh_delete_master.html
